pandas/core/ops/docstrings.py

```diff
@@ -1,7 +1,19 @@
 """
 Templating for ops docstrings
 """
-from typing import Dict, Optional
+from typing import TYPE_CHECKING, Dict, Optional
+
+if TYPE_CHECKING:
+    from mypy_extensions import TypedDict
+
+    class _OpDescriptionsBase(TypedDict):
+        op: str
+        desc: str
+
+    class _OpDescriptions(_OpDescriptionsBase, total=False):
+        reverse: Optional[str]
+        series_examples: Optional[str]
+        df_examples: Optional[str]
```

Comment:
subject to the usual "im happy to defer to you on this" caveat, what is the benefit of defining all this?

Reply:
gives additional type checking but as usual can get by without it.

the `op` and `desc` keys are mandatory, but the current type `Dict[str, Optional[str]]` does not represent that.

In cases like this it may require an `assert op is not None` or `assert desc is not None` to be added without the TypedDict.

Not sure without additional checks, since I'm breaking bits off the other PR and not covered all the untyped defs yet.

In cases where a dictionary is used with specific keys holding different types then more asserts would probably be required.

In this case the values are all strings (or None), we could get by without using a TypedDict